superset/db_engine_specs.py

```diff
@@ -104,15 +104,32 @@ def apply_limit_to_sql(cls, sql, limit, database):
             )
             return database.compile_sqla_query(qry)
         elif LimitMethod.FORCE_LIMIT:
-            no_limit = re.sub(r"""
+            sql_without_limit = cls.get_query_without_limit(sql)
+            return '{sql_without_limit} LIMIT {limit}'.format(**locals())
+        return sql
+
+    @classmethod
+    def get_limit_from_sql(cls, sql):
+        limit_pattern = re.compile(r"""
```

Comment:
I realize that this logic existed previously but shouldn't we use something like `sqlparse` to obtain the limit rather than using a regex?

Reply:
I explored the sqlparse option but there were no nice way to just get the limit without recursively parsing through the query.

Reply:
Isn't this the exact same regex as the one at line `126`? Why duplicating this?